natapokie/ai-fashion-mirror
# install dev dependency husky (testing branch for stacked changes)

## Description

**Fixes #2 and adds on to #20 **

(make sure to review #20 before reviewing this pr!)

Adding pre-commit hook with Husky (runs eslint and prettier) which formats code and checks for errors before committing.
adding lint-staged, makes sure that eslint and prettier are only run for files that are staged and not the entire repo.

Make sure to run `npm install` in root to use these changes

Also side note... to create stacked prs, you can set the `base` to the branch that you did your `git checkout -b` step from.
For example, this branch `2-husky-pre-commit-hooks` is a branch created from `2-add-prettiereslint-for-typescript` -- setting the base to `2-add-prettiereslint-for-typescript` (and not main) only shows the commits from this branch and not the branch i checked out from.
![image](https://github.com/user-attachments/assets/a63c00cd-9f74-4871-bea3-1382a374178a)

## Documentation Used

- [ ] [setting up husky](https://typicode.github.io/husky/)
- [ ] [lint-staged](https://www.npmjs.com/package/lint-staged)

## How To Test?

When you make a commit, you'll see something like the following, if Husky returns no errors, commit was successful,
![image](https://github.com/user-attachments/assets/c825d571-51c1-4052-b168-a2002dd7c0b7)
